### PR TITLE
samples: added foc controller sample for the simple foc shield

### PR DESCRIPTION
**Pull request by uLipe**
Signed-off-by: Felipe Neves <felipe.neves@linaro.org>

### CODE REVIEW

**Comment by uLipe** [comment]
@microbuilder PTAL

Since shields are different I created two variants and set the pinmux according to the used shield

**Comment by uLipe** [comment]
@microbuilder PTAL these fixes are important to make the FoC controller work in both shields.

Also I'm planning to make FoC controller common across shields, since this is expansible with future position plus velocity control.

**Comment by microbuilder** [comment]
@uLipe Can you add the U585 as a board option here? It will make it easier for everyone to test. Thanks.

File: samples/foc_controller_simple_foc_shield/README.rst [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+that is it, no current control in closed loop, instead the pair
+of voltages, direct and quadrature are applied directly to the
+FoC engine, then converted to a 3-phase stator voltages.
+
+The STeP pipeline is triggered each time the motor rotor position
+is measured, then the other node chains are executed, one for 
+execute FoC followed by the PWM Inverter update.
+
+Requirements
+************
```
Comment (microbuilder) [inline samples/foc_controller_simple_foc_shield/README.rst:21]: We should document the hardware assumptions here for the Simple FOC shield, and some notes to help people select an appropriate motor type to test with.

**Comment by uLipe** [comment]
@microbuilder  there is no pwm pin for one of the motor phases, which means extra wiring to the u585 board, the same problem of the stm32 BLDC shield

File: samples/foc_controller_simple_foc_shield/README.rst (hunk 1 above)
Comment (uLipe) [reply samples/foc_controller_simple_foc_shield/README.rst:21]: Just pushed the link for the shield, the relevant information for the motor currently used by the firmware is the voltage link value and its pole pairs.

Do you see any further information needed here?

**Comment by microbuilder** [comment]
> @microbuilder there is no pwm pin for one of the motor phases, which means extra wiring to the u585 board, the same problem of the stm32 BLDC shield

Is the lpcxpresso55s69 the same?

**Comment by uLipe** [comment]
Added DNM here until new publish API gets merged

**Comment by uLipe** [comment]
There is a plan to refactor all the samples and move the prebuilt nodes to a separate folder, so I'm closing this since it will affect all the samples including this one

**Pull request closed without merging**